### ISSUE TITLE
npm module should allow passing extra switches

### ISSUE THREAD

**Issue by jsumners** [ansible/ansible#18795]
_From @jsumners on April 20, 2016 15:15_

##### ISSUE TYPE
- Feature Idea
##### COMPONENT NAME

npm
##### ANSIBLE VERSION

```
% ansible --version                                                                                                 
ansible 2.0.1.0
  config file = /Users/jsumners/Projects/Ansible/banapi-servers/ansible.cfg
  configured module search path = Default w/o overrides
```
##### OS / ENVIRONMENT

OS X - MacPorts
##### SUMMARY

The npm module has a limited subset of switches available to it. This makes some things impossible. For example, annoying modules that require `node-gyp` need to download things from the Internet. Not all systems have a direct, or proxied, connection to the Internet. Therefore, `npm` needs the `--nodedir` switch passed to it as outlined in https://github.com/nodejs/node-gyp/issues/812 . This is not possible as the `npm` module currently exists.

_Copied from original issue: ansible/ansible-modules-extras#2071_

**Comment by jsumners** [comment]
_From @chouseknecht on April 27, 2016 20:48_

@chrishoffman

**Comment by jsumners** [comment]
_From @ansibot on August 4, 2016 19:38_

@chrishoffman, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jsumners** [comment]
_From @ansibot on September 13, 2016 2:48_

@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jsumners** [comment]
_From @ansibot on October 6, 2016 18:3_

@shane-walker, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jsumners** [comment]
_From @ansibot on December 6, 2016 21:52_

This repository has been locked. All new issues and pullrequests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pullrequest to the ansible/ansible repo.

**Comment by Niceplace** [comment]
This issue affects me also.
Proxy configuration is the very example I cannot use the npm module and have to use the command module instead. Even thought it is configured both in the environment variables and system-wide, I still need to pass the proxy via command line options to get npm to correctly fetch packages.

**Comment by ansibot** [comment]
cc @xcambar
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by xcambar** [comment]
@jsumners This issue has been open more than a year ago. It should have been answered, sorry it has never been the case.

Is it still relevant to you? Can you test the behaviour against the ansible 2.4.x series, please? If you can reproduce against these, I'll be more than willing to investigate further on.

**Comment by jsumners** [comment]
@xcambar I don't see anything on https://docs.ansible.com/ansible/latest/npm_module.html that would indicate this has been solved.

The issue is about needing to supply arbitrary switches to the `npm` binary. This module covers the most common ones superbly, but it is not useful in odd circumstances.

**Comment by xcambar** [comment]
I've been reminded of this PR: https://github.com/ansible/ansible/pull/30318

What do you think? Can you test it?

**Comment by jsumners** [comment]
Unfortunately, no. I don't have the time to pull in PRs and put together an environment to test. But I don't see any reason that wouldn't solve the problem.

**Comment by xcambar** [comment]
Okay then I'll put this on hold while the discussion is ongoing at #30318 

Depending on the outcome, I'll point you to the right direction. Keep an eye on the PR!

**Comment by ansibot** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

* lib/ansible/modules/packaging/language/npm.py -> https://galaxy.ansible.com/community/general

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
